packages/headless/src/features/facets/automatic-facet-set/automatic-facet-set-slice.ts

```diff
@@ -24,8 +24,9 @@ export const automaticFacetSetReducer = createReducer(
           state.set[response.field] = {response};
         });
       })
-      .addCase(setDesiredCount, (state, action) => {
-        state.desiredCount = action.payload;
+      .addCase(setOptions, (state, action) => {
```

Comment:
One thing we should do for this action is make each option not mandatory, and if not provided we keep the previous value in store.

ie:

```typescript
// desiredCount=3, numberOfValues=6

dispatch({desiredCount: 5})

// desiredCount=5, numberOfValues=6
```
```